app/src/androidTest/java/com/github/se/cyrcle/di/mocks/MockPermissionHandler.kt

```diff
@@ -1,16 +1,26 @@
 package com.github.se.cyrcle.di.mocks
 
-import android.app.Activity
-import com.github.se.cyrcle.permission.PermissionHandlerInterface
+import com.github.se.cyrcle.MainActivity
+import com.github.se.cyrcle.permission.PermissionHandler
 import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
 
-/** A mock implementation of the [PermissionHandlerInterface] interface. */
-class MockPermissionHandler @Inject constructor() : PermissionHandlerInterface {
+/** A mock implementation of the [PermissionHandler] interface. */
+class MockPermissionHandler @Inject constructor() : PermissionHandler {
+
+  var authorizeLoc: MutableStateFlow<Boolean> = MutableStateFlow(false)
 
   /**
    * Initializes the permissions handler.
    *
-   * @param activity the activity
+   * @param mainActivity the activity
    */
-  override fun initHandler(activity: Activity) {}
+  override fun initHandler(mainActivity: MainActivity) {}
+
+  override fun requestPermission(permissionName: Array<String>) {}
+
+  override fun getPositionState(): StateFlow<Boolean> {
```

Comment:
`GetPermissionState` would be more fitting here because as it is the name is misleading. Seems more like the getter for the user location than for the user location permissions.

Reply:
Yes good comment! `getPermissionState` is maybe a bit general since we could maybe add other permissions later. What do you think of `getLocalisationPerm`?

Reply:
That's perfect for me thanks :+1: